drivers/dma/dma_stm32.c

```diff
@@ -711,10 +697,12 @@ const struct dma_stm32_config dma_stm32_config_##index = {		\
 		    .enr = DT_INST_CLOCKS_CELL(index, bits) },		\
 	.config_irq = dma_stm32_config_irq_##index,			\
 	.base = DT_INST_REG_ADDR(index),				\
-	DMA_STM32_MEM2MEM_INIT(index)					\
+	IF_ENABLED(CONFIG_DMA_STM32_V1,					\
+	(.support_m2m = DT_INST_PROP(index, st_mem2mem),))		\
```

Comment:
For readability, can you add an indentation ?
```suggestion
	IF_ENABLED(CONFIG_DMA_STM32_V1,					\
		  (.support_m2m = DT_INST_PROP(index, st_mem2mem),))	\
```
Apply everywhere if possible

Reply:
I added one more indentation level as suggested, but some of the line splicing are misaligned. Please let me know if that's ok or if I should reformat that code section.